services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/extensions/dynamodb/mappingclient/model/ReadBatch.java

```diff
@@ -16,109 +16,37 @@
 package software.amazon.awssdk.extensions.dynamodb.mappingclient.model;
 
 import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.stream.Collectors;
 import software.amazon.awssdk.annotations.SdkPublicApi;
-import software.amazon.awssdk.extensions.dynamodb.mappingclient.BatchableReadOperation;
 import software.amazon.awssdk.extensions.dynamodb.mappingclient.MappedTableResource;
 import software.amazon.awssdk.extensions.dynamodb.mappingclient.TableMetadata;
-import software.amazon.awssdk.extensions.dynamodb.mappingclient.operations.GetItem;
 import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
 import software.amazon.awssdk.services.dynamodb.model.KeysAndAttributes;
 
-/**
- * Encapsulates a single write transaction that can form a list of transactions that go into a
- * {@link BatchGetItemEnhancedRequest}.
- * Example:
- *
- * {@code
- * ReadBatch.create(myTable, putItem(myItem));
- * ReadBatch.createf(myTable, deleteItem(Key.create(stringValue("id123"))));
- * }
- *
- * @param <T> The type of object this transaction applies to. Can be safely erased as it's not needed outside the
- *            class itself.
- */
 @SdkPublicApi
-public class ReadBatch<T> {
-    private final MappedTableResource<T> mappedTableResource;
-    private final List<BatchableReadOperation> readOperations;
-
-    private ReadBatch(Builder<T> builder) {
-        this.mappedTableResource = builder.mappedTableResource;
-        this.readOperations = Collections.unmodifiableList(builder.readOperations);
-    }
-
-    public static <T> ReadBatch<T> create(MappedTableResource<T> mappedTableResource,
-                                      Collection<BatchableReadOperation> readOperations) {
-        return new Builder<T>().mappedTableResource(mappedTableResource).readOperations(readOperations).build();
-    }
+public class ReadBatch {
+    private final String tableName;
+    private final KeysAndAttributes keysAndAttributes;
 
-    public static <T> ReadBatch<T> create(MappedTableResource<T> mappedTableResource,
-                                      BatchableReadOperation... readOperations) {
-        return new Builder<T>().mappedTableResource(mappedTableResource).readOperations(readOperations).build();
+    private ReadBatch(String tableName, KeysAndAttributes keysAndAttributes) {
```

Comment:
That's the pattern used elsewhere, but in this case it doesn't work because 
1. We're generating new objects in build() that aren't fields in `Builder`
2. We want to keep ReadBatch not use generic type `<T>` which precludes the use of `MappedTableResource` when instantiating `ReadBatch`

Reply:
Introduced a Builder interface as discussed.